test-butler-app/src/main/java/com/linkedin/android/testbutler/ShowImeWithHardKeyboardHelper.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2016 LinkedIn Corp.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.android.testbutler;
+
+import android.content.ContentResolver;
+import android.provider.Settings;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+/**
+ * A helper class for setting the ime keyboard
+ *
+ * This can be used to force the hardware ime keyboard, preventing most occurences of the software keyboard showing on
+ * the screen in tests
+ *
+ * Requires API 22+
+ */
+public class ShowImeWithHardKeyboardHelper {
+
+    private static final String TAG = ShowImeWithHardKeyboardHelper.class.getSimpleName();
+
+    // The constant in Settings.Secure is marked with @hide, so we can't use it
+    private static final String SOFT_KEYBOARD_SETTING = "show_ime_with_hard_keyboard";
+
+    private boolean originalSoftKeyboardMode;
+
+    /**
+     * Should be called before starting tests, to save original ime keyboard state
```

Comment:
nitpick: `IME (Input Method Editor)` is a superset of `keyboard`, so `ime keyboard` is redundant 😉